### PR TITLE
Lab test and family card integration

### PR DESCRIPTION
**Pull request by shashankJTG**
Tasks completed:
1. Lab test card Add/update integration
2. Family Detail card add/update integration
3. Minor fixes

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -88,5 +88,37 @@ export const RELATIONSHIP_OPTIONS = [
...
+        "id": 5,
+        "name": "Presumptive Positive",
+        "theme": "danger-light"
+    },
+    {
+        "id": 6,
+        "name": "Test Inconclusive",
+        "theme": "primary"
+    }
+  ];
```
Comment (shivamjtg) [inline src/constants.js:122]: Formatting issue.

File: src/containers/FacilitiesList/index.js [hunk 2]
```diff
@@ -125,7 +136,7 @@ export function FacilitiesList(props) {
             options.ordering = orderingParam;
         }
         fetchFacilityList(options);
-    }, [queryParams, offset, fetchFacilityList, orderingParam, selectedParams]);
+    }
```
Comment (shivamjtg) [inline src/containers/FacilitiesList/index.js:139]: Semi colon

File: src/containers/Patient/PortieDetails/index.js [hunk 3]
```diff
@@ -52,31 +54,27 @@ export function PortieDetails(props) {
     let portie_response;
     initial['patient'] = patientId;
     if (editable === 'new') {
-      setEditable('');
       portie_response = await createPortieDetails(initial);
       if (portie_response.status) {
+        setEditable('');
         profile.unshift(data);
         createToastNotification(
             ToastUtils.toastDict((new Date()).getTime(), "Added", "Successfully added ", SUCCESS)
```
Comment (shivamjtg) [inline src/containers/Patient/PortieDetails/index.js:62]: Translations  missing.

File: src/containers/Patient/PortieDetails/index.js [hunk 4]
```diff
@@ -52,31 +54,27 @@ export function PortieDetails(props) {
...
       }
     }
     else {
       profile[editable] = data;
-      setEditable('');
       portie_response = await updatePortieDetails(initial, data.id);
       if (portie_response.status) {
+        setEditable('');
         createToastNotification(
             ToastUtils.toastDict((new Date()).getTime(), "updated", "Successfully updated ", SUCCESS)
```
Comment (shivamjtg) [inline src/containers/Patient/PortieDetails/index.js:74]: Translations.

File: src/containers/Patient/PortieDetails/createUpdateForm/index.js [hunk 5]
```diff
@@ -48,7 +45,14 @@ export const CreateUpdatePortieDetails = (props) => {
...
                     >
                         {
-                            props => <Form editMode={editMode} porteaUsers={porteaUsers} details={details} {...props} cancelCallback={cancelCallback}/>
+                            props => <Form 
+                            editMode={editMode} 
+                            porteaUsers={porteaUsers} 
+                            details={details} {...props} 
+                            cancelCallback={cancelCallback} 
+                            createPortieErrors={createPortieErrors}
+                            updatePortieErrors={updatePortieErrors}
```
Comment (shivamjtg) [inline src/containers/Patient/PortieDetails/createUpdateForm/index.js:54]: Formatting.

File: src/containers/Patient/PortieDetails/createUpdateForm/form.js [hunk 6]
```diff
@@ -67,6 +69,30 @@ export default function Form(props) {
 
   return (
     <form onSubmit={handleSubmit}>
+        {
+        createPortieErrors && createPortieErrors.patient &&
```
Comment (shivamjtg) [inline src/containers/Patient/PortieDetails/createUpdateForm/form.js:73]: Formatting issues.

File: src/containers/Patient/LabTestDetails/createUpdateForm/form.js [hunk 7]
```diff
@@ -34,29 +34,26 @@ export default function Form(props) {
...
+  useEffect(() => {
+    if(values.result && (values.result === 3 || values.result === 4)) {
+        setCompeleted(true)
+    } else {
+        setCompeleted(false);
     }
-  }
+  }, [values.result])
+
+  console.log(errors);
```
Comment (gauravksingh) [inline src/containers/Patient/LabTestDetails/createUpdateForm/form.js:56]: remove this.

File: src/containers/Patient/LabTestDetails/createUpdateForm/form.js [hunk 8]
```diff
@@ -34,29 +34,26 @@ export default function Form(props) {
...
-    }
-  }
+  const [completed, setCompeleted] = React.useState(false);
 
-  const onSelectLab = (event, value) => {
-    setFieldValue("testing_lab", value.id);
-    if(saveLabDetails){ 
-      saveLabDetails("testing_lab", value.id);
+  useEffect(() => {
+    if(values.result && (values.result === 3 || values.result === 4)) {
```
Comment (gauravksingh) [inline src/containers/Patient/LabTestDetails/createUpdateForm/form.js:49]: use indexOf >= 0 of list. i.e [3, 4].indexOf(values.result) >= 0.

**Pull request merged**